From 5bdc74235ed0093aea9ee62f6580ce19c59827c6 Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Wed, 12 Nov 2014 15:00:05 +0100
Subject: [PATCH 0057/1800] Odd cross line when drawing an ECalendar with
 multiple columns

Multiple columns mean multiple months on a row, which can be seen
for example in a Recurrence definition dialog when editing an event.
The line was meant as a divider between months, but it had flipped
x and y destination coordinates. After fixing that the line looked
rather distracting, thus I removed it completely.
---
 e-util/e-calendar-item.c | 22 +---------------------
 1 file changed, 1 insertion(+), 21 deletions(-)

diff --git a/e-util/e-calendar-item.c b/e-util/e-calendar-item.c
index 8121c75e96..b415870ee7 100644
--- a/e-util/e-calendar-item.c
+++ b/e-util/e-calendar-item.c
@@ -1070,7 +1070,7 @@ e_calendar_item_draw (GnomeCanvasItem *canvas_item,
 	ECalendarItem *calitem;
 	GtkWidget *widget;
 	GtkStyleContext *style_context;
-	gint char_height, row, col, row_y, bar_height, col_x;
+	gint char_height, row, col, row_y, bar_height;
 	PangoContext *pango_context;
 	PangoFontMetrics *font_metrics;
 	GdkRGBA bg_color;
@@ -1164,26 +1164,6 @@ e_calendar_item_draw (GnomeCanvasItem *canvas_item,
 		gtk_style_context_restore (style_context);
 
 		for (col = 0; col < calitem->cols; col++) {
-			if (col != 0) {
-				col_x = calitem->x1 + calitem->x_offset
-					+ calitem->month_width * col;
-
-				gtk_style_context_save (style_context);
-				gtk_style_context_add_class (
-					style_context,
-					GTK_STYLE_CLASS_SEPARATOR);
-				cairo_save (cr);
-				gtk_render_line (
-					style_context, cr,
-					(gdouble) col_x - 1 - x,
-					(gdouble) row_y + border.top + 1 - y,
-					(gdouble) row_y + bar_height -
-						border.bottom - 2 - y,
-					(gdouble) col_x - x);
-				cairo_restore (cr);
-				gtk_style_context_restore (style_context);
-			}
-
 			e_calendar_item_draw_month (
 				calitem, cr, x, y,
 				width, height, row, col);
-- 
2.15.0

